tools/licensegenerator/src/main.cpp

```diff
@@ -174,6 +242,17 @@ int main(int argc, char* argv[])
 	if (!CommandLine::parse(argc, argv, commandLine))
 		return -1;
 
+	// Set signing scheme
+	std::string signingScheme;
+	if (!commandLine.mSignScheme.empty())
+	{
+		signingScheme = commandLine.mSignScheme;
+		if (!validateSigningScheme(signingScheme))
+			return -1;
+	}
+	else
+		signingScheme = "RSASS_PKCS1v15_SHA1";
```

Comment:
Since this is the default signing scheme, maybe declare it at the top of the file as `constexpr` and include it as first entry in the list of available options?

```
constexpr const char* defaultSigningSceme = "RSASS_PKCS1v15_SHA1";
constexpr const char* supportedSigningSchemes[] =
{
	defaultSigningSceme,
	"RSASS_PKCS1v15_SHA224",
	"RSASS_PKCS1v15_SHA256",
	"RSASS_PKCS1v15_SHA384",
	"RSASS_PKCS1v15_SHA512"
};
```

```
// Set signing scheme
std::string signingScheme = defaultSigningSceme;
if (!commandLine.mSignScheme.empty())
{
	signingScheme = commandLine.mSignScheme;
	if (!validateSigningScheme(signingScheme))
		return -1;
}
```

Reply:
That looks much cleaner. Thanks!